pkg/services/validationService.go

```diff
@@ -185,14 +188,22 @@ func (vs *ValidationService) Validate(admissionReviewReq *admission.AdmissionRev
 		}
 
 		didFailCurrentPolicyCheck := evaluationSummary.PassedPolicyCheckCount == 0
-		if didFailCurrentPolicyCheck && vs.State.GetIsEnforceMode() {
+		shouldBypassByPermissions := vs.shouldBypassByPermissions(resourceUserInfo)
+
+		if didFailCurrentPolicyCheck && vs.State.GetIsEnforceMode() && !shouldBypassByPermissions {
 			allowed = false
 
 			sb.WriteString("\n---\n")
 			sb.WriteString(resultStr)
 		}
 
-		if !vs.State.GetIsEnforceMode() {
+		if shouldBypassByPermissions {
+			if didFailCurrentPolicyCheck {
```

Comment:
can you combine those if's?

Reply:
I thought about it also but I figured out it's problematic bc `if shouldBypassByPermissions && didFailCurrentPolicyCheck` means that it will warn also if `shouldBypassByPermissions is true` and `didFailCurrentPolicyCheck is false` which is not needed